src/SFA.DAS.EmployerIncentives.Api/AppStart/ServiceRegistrationExtension.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using SFA.DAS.EmployerIncentives.Configuration;
+using SFA.DAS.EmployerIncentives.Infrastructure.Api;
+using SFA.DAS.EmployerIncentives.Interfaces;
+using SFA.DAS.EmployerIncentives.Services;
+
+namespace SFA.DAS.EmployerIncentives.Api.AppStart
+{
+    public static class ServiceRegistrationExtension
+    {
+        public static IServiceCollection AddDasHttpClientsAndAssociatedServices(this IServiceCollection services, IConfiguration configuration, IWebHostEnvironment env)
+        {
+            services.AddTransient(typeof(ManagedIdentityApiHandler));
+            services.AddTransient<IRestApiClient, RestApiClient>();
+
+            var httpBuilder = services.AddHttpClient("EmployerIncentivesInnerApi", c =>
+                {
+                    var apiConfig = GetConfigSection(configuration, EmployerIncentivesConfigurationKeys.EmployerIncentivesInnerApiConfiguration);
+                    c.BaseAddress = new Uri(apiConfig.Url);
+                })
+                .AddTypedClient<IEmployerIncentivesPassThroughService, EmployerIncentivesPassThroughService>();
+
+            if (!env.IsDevelopment())
```

Comment:
This method can be a PITA!  If you want to run the outer locally but with the inner from AT it won't work.  I've always found IsLoopback on the API base address to be easier.

Reply:
Don't think IsLoopback works in core. I used in framework, but found I couldn't implement it in Core.   But point taken about mixing environments

Reply:
These extension methods will be consolidated into the shared project in the near future